test/e2e/resources/components/mocks/functions/main.brs

```diff
@@ -61,6 +61,27 @@ sub main()
     mock.clearMock()
     print mock.calls.count() ' => 0
     print mock.results.count() ' => 0
+
+    mock.mockReturnValue("foo")
+    spyOnMe(1234)
+    print mock.calls[0][0] ' => 1234
+    print mock.results[0] ' => "foo"
```

Comment:
I was going to ask if this matched the `jest.fn()` API and clearly you already thought of that — this is near-identical!  I think the only thing we might want to consider is making `results` entries associative arrays of `{ type: "return" | "throw", value: dynamic }`.  That'll match Jest closer but also makes supporting `try/catch` a bit easier when that happens 😃

Reply:
Great idea! When we first implemented the mocks, `try/catch` was not yet a Brightscript feature, which is why we didn't include a `type`. This will be a breaking change, though, and is not strictly related to the improvements in this PR -- can we do that in a separate PR?

Reply:
> @lkipke Great idea! When we first implemented the mocks, `try/catch` was not yet a Brightscript feature, which is why we didn't include a `type`. This will be a breaking change, though, and is not strictly related to the improvements in this PR -- can we do that in a separate PR?

Sure thing!  I saw you opened #624 so that works for me 😃